src/v2/view-builder/views/UserVerificationDeviceChallengePollView.js

```diff
@@ -0,0 +1,159 @@
+      case 'LOOPBACK':
+        this.title = loc('signin.with.fastpass', 'login');
+        this.add('<div class="spinner"></div>');
+        this.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+        break;
+      case 'CUSTOM_URI':
+        this.title = loc('customUri.title', 'login');
+        this.subtitle = loc('customUri.subtitle', 'login');
+        this.add(`
+          {{{i18n code="customUri.content" bundle="login"}}}
+        `);
+        this.customURI = deviceChallenge.href;
+        this.doCustomURI();
+        break;
+      case 'UNIVERSAL_LINK':
+        this.title = loc('universalLink.userVerification.title', 'login');
+        this.add(`
+          {{i18n code="universalLink.userVerification.content.p1" bundle="login"}}
+        `);
+        this.add(`
+          {{i18n code="universalLink.userVerification.content.p2" bundle="login"}}
+        `);
+        this.add(createButton({
+          className: 'ul-button button button-wide button-primary',
+          title: loc('universalLink.userVerification.button', 'login'),
+          click: () => {
+            // only window.location.href can open universal link in iOS/MacOS
+            // other methods won't do, ex, AJAX get or form get (Util.redirectWithFormGet)
+            Util.redirect(deviceChallenge.href);
+          }
+        }));
+      }
+    },
+
+    doLoopback (authenticatorDomainUrl = '', ports = [], challengeRequest = '') {
+      let currentPort;
+      let foundPort = false;
+      let countFailedPorts = 0;
+
+      const getAuthenticatorUrl = (path) => {
+        return `${authenticatorDomainUrl}:${currentPort}/${path}`;
+      };
+
+      const checkPort = () => {
+        return request({
+          url: getAuthenticatorUrl('probe'),
+          // in loopback server, SSL handshake sometimes takes more than 1000 ms and thus needs additional timeout
+          // however, increasing timeout is a temporary solution since user will need to wait much longer in worst case
+          // TODO: OKTA-278573 Android timeout is temporarily set to 3000ms and needs optimization post-Beta
+          timeout: DeviceFingerprint.isAndroid() ? 3000 : 1000
+        });
+      };
+
+      const onPortFound = () => {
+        foundPort = true;
+        return request({
+          url: getAuthenticatorUrl('challenge'),
+          method: 'POST',
+          data: JSON.stringify({ challengeRequest }),
+          timeout: 3000 // authenticator should respond within 3000ms for challenge request
+        });
+      };
+
+      const onFailure = () => {
+        Logger.error(`Something unexpected happened while we were checking port ${currentPort}.`);
+      };
+
+      const doProbing = () => {
+        return checkPort()
+          // TODO: can we use standard ES6 promise methods, then/catch?
+          .done(onPortFound)
+          .fail(onFailure);
+      };
+
+      let probeChain = Promise.resolve();
```

Comment:
Since we're using jquery.ajax to make request, would you consider to switch to `jquery.Deferred`? just to be consistent.